docs/api-reference/next/router.md

```diff
@@ -138,6 +136,62 @@ export default function Page() {
 }
 ```
 
+#### Resetting state after navigation
+
+When navigating to the same page in Next.js, the page's state **will not** be reset by default:
+
+```jsx
+// pages/[pageTitle].jsx
+
+import Link from "next/link"
+import { useState } from "react"
+
+export const getServerSideProps = async ({ params }) => ({
+  props: {
+    pageTitle: params["pageTitle"]
+  }
+})
+
+export default function Page({ pageTitle }) {
+  const [title] = useState(pageTitle)
+  return (
+    <div>
+      <h1>{title}</h1>
+      <Link href="/one">
+        <a>one</a>
+      </Link>{" "}
+      <Link href="/two">
+        <a>two</a>
+      </Link>
+    </div>
+  )
+}
+```
+
+Above, navigating between `/foo` and `/bar` **will not** change the page title. The `useState` is initialized for the first `Page` component, and not re-initialized when navigating to a different `Page` instance. This is because the top-level React component, `Page`, is the same.
```

Comment:
```suggestion
In the above example, navigating between `/one` and `/two` **will not** reset the count . The `useState` is maintained between renders because the top-level React component, `Page`, is the same.
```